resources/js/components/dashboard/components/IndividualSponsor.tsx

```diff
@@ -0,0 +1,351 @@
+import React, { Component } from 'react';
+import {withRouter, RouteComponentProps} from 'react-router';
+import { Page, Card, SkeletonBodyText, Image, MediaCard, Thumbnail, Layout, Heading, TextContainer, DisplayText, Button, Badge, Modal, Stack, RangeSlider, KeyboardKey, Spinner } from '@shopify/polaris';
+import Dashboard404 from '../Dashboard404';
+import { ISponsor, IUserDetails } from '../../../interfaces/dashboard.interfaces';
+import axios from 'axios';
+import { toast } from 'react-toastify';
+import md5 from 'md5';
+import { textFieldQuestions } from './Apply';
+import Linkify from 'linkifyjs/react';
+import { IAssetInformation, IResourceDefinition, ISponsorData } from '../../../interfaces/sponsors.interfaces';
+import "react-responsive-carousel/lib/styles/carousel.min.css"; // requires a loader
+import { Carousel } from 'react-responsive-carousel';
+import { portal } from '@shopify/polaris/dist/types/latest/src/components/shared';
+
+interface IIndividualSponsorProps {
+    SponsorId: string,
+}
+
+interface IIndividualSponsorState {
+    SponsorId: number | undefined,
+    loading: boolean,
+    Sponsor: ISponsor | undefined,
+    user: IUserDetails | undefined,
+    loadingDefinitions: boolean,
+    resources: undefined,
+    totalNo: number,
+    portalInfo: undefined,
+}
+
+const sponsorFields: { id: string, title: string, placeholder: string }[] = [
+    { id: "1", title: "Industry", placeholder: "" },
+    { id: "2", title: "Homepage", placeholder: "Mention anything you want -- it doesn’t have to be technology-related!" },
+    { id: "3", title: "Description", placeholder: "" },
+    { id: "4", title: "Opportunities", placeholder: "For example GitHub, LinkedIn or your website. Put each link on a new line. " },
+]
+
+class IndividualSponsor extends Component<IIndividualSponsorProps & RouteComponentProps, IIndividualSponsorState> {
+    
+    state = {
+        SponsorId: undefined, // possibly redundant copy of ID
+        loading: true, // loading main spons object
+        loadingDefinitions: true, // loading additional resource cards
+        Sponsor: undefined, // key sponsor object
+        user: undefined,
+        resources: undefined, //where all resource cards are stored
+        totalNo: 0, //for pagination purposes
+        portalInfo: {
+            data:{
+                ["description"]:"",
+                ["url"]:"",
+                ["discord invite link"]:"",
+            },
+            files:[],
+        }, //for company main info purposes
+    }
+
+    constructor(props: IIndividualSponsorProps & RouteComponentProps){
+        super(props);
+    }
```

Comment:
We can skip this - it is the default one.

Reply:
Removed